Lab_2_3_4_5/src/Grammar/Main.java

```diff
@@ -1,10 +1,26 @@
 package Grammar;
 
+import java.io.FileOutputStream;
+import java.io.PrintStream;
+
 public class Main {
     public static void main(String[] args) throws Exception {
-        var grammar = new Grammar("src\\g1.txt");
-        var parser = new ParserLR0(grammar);
-        parser.printGotoTable();
-        parser.printActionTable();
+        PrintStream out = new PrintStream(
+                new FileOutputStream("result.txt", false), true);
+        System.setOut(out);
+//        {
```

Comment:
maybe reading a value and an if would have been more elegant, so that you don't have commented lines that are valid lines of code